src/backend/distributed/utils/attribute.c

```diff
@@ -619,3 +604,128 @@ MultiTenantMonitorshmemSize(void)
 
 	return size;
 }
+
+
+/*
+ * ExtractTopComment extracts the top-level multi-line comment from a given input string.
+ */
+static char *
+ExtractTopComment(const char *inputString)
+{
+	int commentStartCharsLength = 2;
+	int inputStringLen = strlen(inputString);
+	if (inputStringLen < commentStartCharsLength)
+	{
+		return NULL;
+	}
+
+	int commentEndCharsIndex = 0;
+
+	/* If query doesn't start with a comment, return NULL */
+	if (inputString[commentEndCharsIndex] != '/' ||
+		inputString[commentEndCharsIndex + 1] != '*')
+	{
+		return NULL;
+	}
+
+	/* Skip the comment start characters */
+	commentEndCharsIndex += commentStartCharsLength;
+	
+	/* Find the first comment end character */
+	while (commentEndCharsIndex < inputStringLen &&
+		   !(inputString[commentEndCharsIndex] == '*' &&
+			 inputString [commentEndCharsIndex + 1] == '/'))
+	{
+		commentEndCharsIndex++;
+	}
+
+	/* If there is no end of comment chars , return NULL */
+	if (inputString[commentEndCharsIndex] != '*' &&
+		inputString[commentEndCharsIndex + 1] != '/')
+	{
+		return NULL;
+	}
+
+	if (commentEndCharsIndex > commentStartCharsLength)
+	{
+		return Substring(inputString, commentStartCharsLength, commentEndCharsIndex);
+	}
+	else
+	{
+		return NULL;
+	}
+}
+
+
+/* Extracts a substring from the input string between the specified start and end indices.*/
+static char *
+Substring(const char *str, int start, int end)
+{
+	int len = strlen(str);
+
+	/* Ensure start and end are within the bounds of the string */
+	if (start < 0 || end > len || start > end)
+	{
+		return NULL;
+	}
+
+	/* Allocate memory for the substring */
+	char *substr = (char *) palloc((end - start + 1) * sizeof(char));
+
+	/* Copy the substring to the new memory location */
+	strncpy_s(substr, end - start + 1, str + start, end - start);
+
+	/* Add null terminator to end of the substring */
+	substr[end - start] = '\0';
+
+	return substr;
+}
+
+
+/*  EscapeCommentChars adds a backslash before each occurrence of '*' or '/' in the input string */
+static char *
+EscapeCommentChars(const char *str)
+{
+	int originalStringLength = strlen(str);
+	char *escapedString = (char *) palloc0((originalStringLength * 2 + 1) * sizeof(char));
+	int escapedStringIndex = 0;
+
+	for (int originalStringIndex = 0; originalStringIndex < originalStringLength;
+		 originalStringIndex++)
+	{
+		if (str[originalStringIndex] == '*' || str[originalStringIndex] == '/')
```

Comment:
I feel like we should also escape the escape character
```suggestion
		if (str[originalStringIndex] == '*' || str[originalStringIndex] == '/' || str[originalStringIndex] == '\\')
```

Reply:
Actually probably it's probably not needed because of the way we do unescaping. Some tests are definitely needed with a backslash in the tenant ID though.